pkg/resource/parameter_group/update_pre_build.go

```diff
@@ -0,0 +1,73 @@
+package parameter_group
+
+import (
+	"context"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+)
+
+func (rm *resourceManager) resetParameterGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (updated *resource, err error) {
+	if (desired.ko.Spec.ParameterNameValues == nil || len(desired.ko.Spec.ParameterNameValues) == 0) && latest.ko.Spec.ParameterGroupName != nil {
```

Comment:
`Spec.ParameterGroupName` is a required field, and is the primary identifier. I don't believe we need to check for `nil` anywhere for this particular property.